src/titan_checkpoint_test.cc

```diff
@@ -0,0 +1,549 @@
+#include <thread>
+#include "titan/checkpoint.h"
+#include "titan/db.h"
+
+#include "port/port.h"
+#include "port/stack_trace.h"
+#include "rocksdb/env.h"
+#include "test_util/fault_injection_test_env.h"
+#include "test_util/sync_point.h"
+#include "test_util/testharness.h"
+#include "test_util/testutil.h"
+
+namespace rocksdb {
+namespace titandb {
+
+class CheckpointTest : public testing::Test {
+ protected:
+  // Sequence of option configurations to try
+  enum OptionConfig {
+    kDefault = 0,
+  };
+  int option_config_;
+
+ public:
+  std::string dbname_;
+  std::string alternative_wal_dir_;
+  Env* env_;
+  TitanDB* db_;
+  TitanOptions last_options_;
+  std::vector<ColumnFamilyHandle*> handles_;
+  std::string snapshot_name_;
+
+  CheckpointTest() : env_(Env::Default()) {
```

Comment:
please add a case that there is a Titan GC finished within the process of creating checkpoint.

Reply:
Done